### ISSUE TITLE
gcloud has stopped working after 7.3.0 update

### ISSUE THREAD

**Issue by andreikondratev** [PowerShell/PowerShell#18529]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I installed 7.3.0 with msi and after that kept getting the error like when executing `gcloud` commands
```
google-cloud-sdk\platform\bundledpython\python.exe: can't find '__main__' module in ''
```

Tried to reinstall gcloud (with and without bundled python) and powershell a few times without much luck.

**Works fine after downgrade to PS 7.2.5**

### Expected behavior

```console
PS C:\> gcloud -h
Usage: gcloud [optional flags] <group | command>
  group may be           access-approval | access-context-manager |
                         active-directory | ai | ai-platform | anthos |
                         api-gateway | apigee | app | artifacts | asset |
                         assured | auth | batch | beta | bigtable | billing |
                         bms | builds | certificate-manager | cloud-shell |
                         components | composer | compute | config | container |
                         data-catalog | database-migration | dataflow |
                         dataplex | dataproc | datastore | datastream | debug |
                         deploy | deployment-manager | dns | domains |
                         edge-cloud | emulators | endpoints |
                         essential-contacts | eventarc | filestore | firebase |
                         firestore | functions | game | healthcare | iam | iap |
                         identity | ids | iot | kms | logging | memcache |
                         metastore | ml | ml-engine | monitoring |
                         network-connectivity | network-management |
                         network-security | network-services | notebooks |
                         org-policies | organizations | policy-intelligence |
                         policy-troubleshoot | privateca | projects | pubsub |
                         recaptcha | recommender | redis | resource-manager |
                         resource-settings | run | scc | scheduler | secrets |
                         service-directory | services | source | spanner | sql |
                         storage | tasks | topic | transcoder | transfer |
                         workflows | workspace-add-ons
  command may be         cheat-sheet | docker | feedback | help | info | init |
                         survey | version

For detailed information on this command and its flags, run:
  gcloud --help
```

### Actual behavior

```console
PS C:\> gcloud -h
google-cloud-sdk\platform\bundledpython\python.exe: can't find '__main__' module in ''
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by arifinsiam** [comment]
facing the same issue, have you found any solutions yet?

**Comment by cesarjosue-reyes** [comment]
Same issue

**Comment by al-cheb** [comment]
@SteveL-MSFT, it will be affected -  https://github.com/actions/runner-images

**Comment by al-cheb** [comment]
> facing the same issue, have you found any solutions yet?

There is no any issue with PowerShell Desktop.

**Comment by al-cheb** [comment]
https://github.com/PowerShell/PowerShell/issues/13428#issuecomment-1297723605
Workaround is to set `$PSNativeCommandArgumentPassing` - https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Preference_Variables?view=powershell-7.3#psnativecommandargumentpassing

**Comment by andykcfung** [comment]
Another workaround is to start "cmd" inside powershell, login and then exit the "cmd" shell.

**Comment by andykcfung** [comment]
> > facing the same issue, have you found any solutions yet?
> 
> There is no any issue with PowerShell Desktop.

FWIW, "PowerShell Desktop" is a very old version of PowerShell. Run the command "$PSVersionTable" on mine and it shows "PSVersion" to be 5.1.x

**Comment by SteveL-MSFT** [comment]
If someone can confirm that setting `$PSNativeCommandArgumentPassing = "Legacy"` works and `gcloud` works, we can add that to legacy list so that it defaults to previous behavior.  Also, is the actual command `gcloud.exe` or `gcloud.cmd`?  The latter should automatically use legacy passing.

**Comment by SehiiRohoza** [comment]
> If someone can confirm that setting `$PSNativeCommandArgumentPassing = "Legacy"` works and `gcloud` works, we can add that to legacy list so that it defaults to previous behavior. Also, is the actual command `gcloud.exe` or `gcloud.cmd`? The latter should automatically use legacy passing.

Works for me, `gcloud.exe`.

**Comment by daxian-dbw** [comment]
Running gcloud is not running gcloud.exe, but resolved to the gcloud.ps1 file under the installation folder. The gcloud.ps1 file eventually calls python.exe with a generated argument array. For the gcloud -h, the generated argument array is essentially the following array:
```
$arg_array = $('', '-S', 'C:\Users\username\AppData\Local\Google\Cloud SDK\google-cloud-sdk\lib\gcloud.py', '-h')
```
The leading empty string is the problem that cause it fail in the new Standard/Windows mode. With legacy mode, the leading empty string is ignored when passing to python.exe, but with the new standard/windows mode, the empty string is passed in as well, which causes `python.exe` to fail.

**Comment by veleek** [comment]
For anybody who runs into this issue and is looking for a quick workaround.  Open up `gcloud.ps1` (you can get the path using `Get-Command gcloud | Select-Object Source`) and then replace Line 109:

```powershell
$run_args_array += $cloudsdk_python_args.split(' ')
```

with

```powershell
$run_args_array += $cloudsdk_python_args.split(' ', [StringSplitOptions]::RemoveEmptyEntries)
```

**Comment by catosaurusrex2003** [comment]
may not be related to the real problem of this error but i was getting the exact same error and
my error got SOLVED
the problem was that while installing gcloud CLI i installed it just for my user on C drive , whereas i was calling gcloud outside of the user path.
then i opened powershell inside user and then gcloud started to work

**Comment by alteredego** [comment]
I can confirm that this has now been fixed with Google Cloud CLI version 413.0.0
https://issuetracker.google.com/issues/259295558
